salt/modules/file.py

```diff
@@ -2189,7 +2189,7 @@ def replace(path,
                 raise CommandExecutionError("Exception: {0}".format(exc))
             # write new content in the file while avoiding partial reads
             try:
-                fh_ = salt.utils.atomicfile.atomic_open(path, 'w')
+                fh_ = salt.utils.atomicfile.atomic_open(path, 'wb')
```

Comment:
Windows does strange things with line endings if not opened in binary mode. We open in binary when we can on Windows.